README.md

```diff
@@ -83,52 +85,36 @@ app's Gemfile.
 
     The AWS principal whose credentials are in the environment will need to be allowed to assume this role.
 
-### Upgrading from 3.1.0
-
-Versions of Journaled prior to 4.0 relied directly on environment variables for stream names, but now stream names are configured directly.
-When upgrading, you can use the following configuration to maintain the previous behavior:
-
-```ruby
-Journaled.default_stream_name = ENV['JOURNALED_STREAM_NAME']
-```
-
-If you previously specified a `Journaled.default_app_name`, you would have required a more precise environment variable name (substitute `{{upcase_app_name}}`):
+4. Last, but not least, you may (optionally) set up your base `ApplicationController` and `ApplicationJob` classes
+    to tag events with `tags` metadata:
 
-```ruby
-Journaled.default_stream_name = ENV["{{upcase_app_name}}_JOURNALED_STREAM_NAME"]
-```
-
-And if you had defined any `journaled_app_name` methods on `Journaled::Event` instances, you can replace them with the following:
-
-```ruby
-def journaled_stream_name
-  ENV['{{upcase_app_name}}_JOURNALED_STREAM_NAME']
-end
-```
-
-When upgrading from 3.1 or below, `Journaled::DeliveryJob` will handle any jobs that remain in the queue by accepting an `app_name` argument. **This behavior will be removed in version 5.0**, so it is recommended to upgrade one major version at a time.
-
-### Upgrading from 2.5.0
-
-Versions of Journaled prior to 3.0 relied direclty on `delayed_job` and a "performable" class called `Journaled::Delivery`.
-In 3.0, this was superceded by an ActiveJob class called `Journaled::DeliveryJob`, but the `Journaled::Delivery` class was not removed until 4.0.
+    ```ruby
+    class ApplicationController < ActionController::Base
+      around_action do |_controller, action|
+        tags = { request_id: request.request_id, current_user_id: current_user&.id }
+        Journaled.tagged(tags) { action.call }
+      end
```

Comment:
We could also take inspiration from Sentry and allow for something like this:

```ruby
before_action do
  Journaled.set_tags(request_id: request.request_id, current_user_id: current_user&.id)
end
```

The difference there is that `set_tags` would stamp the data onto the current context for its entire duration (until the request ends) -- which is also fine in a lot of cases. The `Journaled.tagged(..., &block)` usage will clear itself more neatly and may be more useful in non-request contexts, but I'm not particularly married to one or the other. (We could ship both?)

Any thoughts?

Reply:
Added a `Journaled.tag!(...)` helper for this!